atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/RetryOnSocketTimeoutExceptionProxy.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.http.v2;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.reflect.AbstractInvocationHandler;
+import com.palantir.common.base.Throwables;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.net.SocketTimeoutException;
+import java.util.function.Supplier;
+
+/**
+ * This proxy exists to retry in case of a SocketTimeoutException (something that isn't retried natively by Dialogue).
+ */
+public final class RetryOnSocketTimeoutExceptionProxy<T> extends AbstractInvocationHandler {
+    private static final int MAX_NUM_RETRIES = 5;
+    private final Supplier<T> delegate;
+
+    private static int numRetries = 0;
```

Comment:
ooooh this looks weird

does FFP use a `static` here?!

Reply:
oh you are right. This shouldn't be static.